src/endpoint/s3/s3_rest.js

```diff
@@ -216,25 +216,35 @@ async function authorize_request_policy(req) {
     }
 
     const account = req.object_sdk.requesting_account;
-    const account_identifier = req.object_sdk.nsfs_config_root ? account.name.unwrap() : account.email.unwrap();
-    const is_system_owner = account_identifier === system_owner.unwrap();
+    const account_identifier_name = req.object_sdk.nsfs_config_root ? account.name.unwrap() : account.email.unwrap();
+    const account_identifier_id = req.object_sdk.nsfs_config_root ? account._id : undefined;
+    const is_system_owner =
+        (account_identifier_name === system_owner.unwrap()) ||
+        (account_identifier_id && account_identifier_id === system_owner.unwrap());
 
     // @TODO: System owner as a construct should be removed - Temporary
     if (is_system_owner) return;
 
     const is_owner = (function() {
         if (account.bucket_claim_owner && account.bucket_claim_owner.unwrap() === req.params.bucket) return true;
         if (req.object_sdk.nsfs_config_root && account._id === owner_account.id) return true; // NC NSFS case
-        if (account_identifier === bucket_owner.unwrap()) return true;
+        if (account_identifier_name === bucket_owner.unwrap()) return true;
```

Comment:
Don't we need to remove this condition for fixing the bucket_policy issue?

Reply:
I don't think so.
What scenario  are you considering?

Generally, the bucket_policy issue is resolved by checking the principal twice - once for account id and once for account_name. If one of them passes, the request is authorized.